wagtail_streamfield_migration_toolkit/management/commands/streamdatamigration.py

```diff
@@ -121,7 +120,10 @@ def handle(self, *args, **options):
         # If the user doesn't give a name for the migration file, generate one based on the
         # operations used.
         if not self.migration_name:
-            self.migration_name = "_".join(self._migration_names)[:30]
+            self.migration_name = "_".join(self._migration_names)[:40]
+
+        elif self.migration_name and not self.migration_name.isidentifier():
```

Comment:
I don't think you need `elif` and `self.migration_name`, since your `if` condition is `not self.migration_name`